### PR TITLE
Add ES7 interop

### LINKED ISSUE #160: Implement [Symbol.observable] for ES7 interop compliance
Hey @briancavalier! I know you and @jhusain and I talked a few months ago about interop between reactive streaming libraries like this and I'm here to offer a little help bridging that gap.

we've been working hard on the next version of [RxJS](http://github.com/ReactiveX/RxJS), and we're getting closer. Part of that work was to try to adhere to the current ES7 Observable spec by @zenparsing. The part of particular interest there for this library is the `[Symbol.observable]()` method.

### Implementing [Symbol.observable]

Really all that is involved with that is to add a `[Symbol.observable]()` method to your streaming type, and  have it return any object with a `subscribe(observer)` method, and a `[Symbol.observable]()` method.

The benefits here are:

- interop between streaming libraries
- makes your type usable with frameworks looking for that contract (Angular 2 is likely to have support for this, for example)
- eventual interop with JavaScript core whenever ES7 Observable lands.

I think a basic implementation might look like:

```js
[Symbol.observable]() {
  var self = this;
  return {
    subscribe(observer) {
      self.observe(observer.next.bind(observer))
        .then(observer.complete.bind(observer), observer.error.bind(observer));
    }
    [Symbol.observable]() { return this; }
  };
}
```

### Consuming "lowercase-o" observables...

On your end, you could implement something inside of your `from` method to handle any type with a `[Symbol.observable]` method on it. As a guineapig you can use `@reactivex/rxjs` over npm, if you choose. Again it's pretty basic. You'd just call `[Symbol.observable]()` then call `subscribe(observer)` on the returned object, where observer has `next`, `error` and `complete` methods.

### Other things...

I've actively tried to make any monadic-bind-esque operator in RxJS Next support objects that implement `Symbol.observable`. This means, effectively that if you were to implement this, people could jump between Most.js and RxJS using flatMaps and switches, etc.  I'll be reaching out to the Bacon.js community with the same information.

### PR DESCRIPTION
**Pull request by briancavalier**
@blesh You inspired me :)  I took a crack at adding interop in both directions.

This is not intended to be a complete implementation of the ES7 spec.  The intention is to provide interoperability between most.js and other libraries that provide minimal ES7 observable compliance, which basically means libs that provide and consume types with a `[Symbol.observable]()` that returns an object with a well-behaved `.subscribe()` method.

### Todo

- [x] Unit tests
- [x] Docs

See #240
Close #160

### CODE REVIEW

File: most.js [hunk 1]
```diff
@@ -23,6 +24,19 @@ exports.from     = from;
 exports.periodic = periodic;
 
 //-----------------------------------------------------------------------
+// ES7 Observable interop
+
+var subscribe = require('./lib/observable/subscribe').subscribe;
+
+Stream.prototype.subscribe = function(subscriber) {
+	subscribe(subscriber, this);
```
Comment (campersau) [inline most.js]: Would it make sense to return the disposable here?

Comment (briancavalier) [inline most.js]: Oh my, yes! Good catch @campersau.

**Comment by blesh** [comment]
This looks awesome, @briancavalier!

What do you think about having some shared repository or something with interop tests between known reactive libraries?

File: most.js [hunk 2]
```diff
@@ -23,6 +24,19 @@ exports.from     = from;
...
+// ES7 Observable interop
+
+var subscribe = require('./lib/observable/subscribe').subscribe;
+
+Stream.prototype.subscribe = function(subscriber) {
+	return subscribe(subscriber, this);
+};
+
+Stream.prototype[symbolObservable] = function() {
+	return this;
```
Comment (blesh) [inline most.js]: <3

**Comment by briancavalier** [comment]
Thanks @blesh!

> shared repository or something with interop tests

I think it's a great idea.  I would happily donate any/all the tests in this PR that seem useful for a shared interop tests repo.

**Comment by TylorS** [comment]
Love the idea of a shared interop test repo :+1:

**Comment by briancavalier** [comment]
Ok, I think this has a reasonably complete set of unit tests for a first go.  I'll add docs over the next few days, and hopefully this will be ready to go.

**Comment by briancavalier** [comment]
Rebased.  I think this is ready!  Gonna look over it once more and wait for travis.

**Comment by TylorS** [comment]
:+1:

**Comment by chicoxyzzy** [comment]
@blesh @briancavalier 
> shared repository or something with interop tests between known reactive libraries

does it exist?

**Comment by briancavalier** [comment]
@chicoxyzzy As far as I know, it doesn't exist yet.  @blesh may know more, though.

**Comment by blesh** [comment]
No it doesn't exist yet

**Pull request merged**